src/theme/typography.ts

```diff
@@ -0,0 +1,119 @@
+import { createTheme } from '@mui/material';
+
+declare module '@mui/material/styles' {
+  interface TypographyVariants {
+    subheader1: React.CSSProperties;
+    subheader2: React.CSSProperties;
+    subheader3: React.CSSProperties;
+    subheader4: React.CSSProperties;
+  }
+
+  // allow configuration using `createTheme`
+  interface TypographyVariantsOptions {
+    subheader1?: React.CSSProperties;
+    subheader2?: React.CSSProperties;
+    subheader3?: React.CSSProperties;
+    subheader4?: React.CSSProperties;
+  }
+}
+
+// Update the Typography's variant prop options
+declare module '@mui/material/Typography' {
+  interface TypographyPropsVariantOverrides {
+    subheader1: true;
+    subtitle1: false;
+    subheader2: true;
+    subtitle2: false;
+    subheader3: true;
+    subheader4: true;
+  }
+}
+
+export const typography = createTheme({
```

Comment:
I am not sure if we need to create a theme only for typography, maybe we need to talk about it.